.pre-commit-config.yaml

```diff
@@ -3,7 +3,11 @@ repos:
     rev: v4.5.0 # must match requirements-tests.txt
     hooks:
       - id: trailing-whitespace
+        # speedup hook by leaving python files to Black & Ruff
+        exclude: .*\.pyi?$
```

Comment:
Black does not necessarily fix trailing whitespace in string literals

Reply:
That would only be multiline string literals right?
Do we even have those in stubs? Given we don't have docstrings (outside maybe generated protobuf files)

After checking, yes we do, in `deprecated` multiline messages, but it seems think Ruff does catch those, although it's considered an "unsafe-fix" (which is true and why black doesn't touch them):
![image](https://github.com/python/typeshed/assets/1350584/405cf2a6-f4dc-47ff-942d-f250439d5985)

That being said, if it was removed anyway by another pre-commit hook, I can just add `W291` to our unsafe fixes hook.

Reply:
On that note, @AlexWaygood any reason not to use https://docs.astral.sh/ruff/settings/#lint_extend-safe-fixes instead of a separate pre-commit hook?

Reply:
no reason other than that I forgot about that option, I guess :)